packages/ui-icons-lucide/src/wrapLucideIcon/index.tsx

```diff
@@ -49,84 +52,30 @@ export function wrapLucideIcon(Icon: LucideIcon): LucideIcon {
       elementRef,
       themeOverride,
       absoluteStrokeWidth,
-      className,
-      style,
       ...rest
     } = props
 
-    const theme = useTheme() as Theme
-    const iconTheme = theme?.newTheme?.components?.Icon
+    // Get icon props from context (if available)
+    const contextProps = useContext(IconPropsContext)
+
+    // Merge props: direct props take precedence over context props
+    const finalSize = size ?? contextProps?.size
+    const finalColor = color ?? contextProps?.color
 
     const handleElementRef = (el: SVGSVGElement | null) => {
       if (typeof elementRef === 'function') {
         elementRef(el)
       }
     }
 
-    // Convert semantic size to pixels for Lucide
-    let numericSize: number | undefined
-    let semanticSize: string | undefined
-    if (typeof size === 'string' && iconTheme) {
-      // Construct theme property name (e.g., 'xs' -> 'sizeXs')
-      const propName = `size${size.charAt(0).toUpperCase()}${size.slice(
-        1
-      )}` as keyof typeof iconTheme
-      if (propName in iconTheme) {
-        // Semantic size token from theme
-        semanticSize = size
-        numericSize = px(iconTheme[propName] as string)
-      }
-    } else if (typeof size === 'number') {
-      numericSize = size
-    }
-
-    // Convert semantic strokeWidth to pixels for Lucide
-    let numericStrokeWidth: number | string | undefined
-    if (typeof strokeWidth === 'string' && iconTheme) {
-      // Construct theme property name (e.g., 'xs' -> 'strokeWidthXs')
-      const propName = `strokeWidth${strokeWidth
-        .charAt(0)
-        .toUpperCase()}${strokeWidth.slice(1)}` as keyof typeof iconTheme
-      if (propName in iconTheme) {
-        // Semantic stroke width token from theme
-        numericStrokeWidth = px(iconTheme[propName] as string)
-      } else {
-        // Not a semantic token, use as-is (custom string value)
-        numericStrokeWidth = strokeWidth
-      }
-    } else {
-      // Numeric value (custom stroke width)
-      numericStrokeWidth = strokeWidth
-    }
-
-    // Determine if color is semantic (theme token) or custom CSS
-    let colorValue: string | undefined
-    let customColor: string | undefined
-    if (color) {
-      if (color === 'inherit') {
-        colorValue = color
-      } else if (
-        iconTheme &&
-        color in iconTheme &&
-        !color.startsWith('size') &&
-        !color.startsWith('strokeWidth') &&
-        color !== 'dark'
-      ) {
-        // Semantic color token from theme (exclude size/strokeWidth/dark properties)
-        colorValue = color
-      } else {
-        // Custom CSS color (e.g., "#ff0000", "rgb(255, 0, 0)")
-        customColor = color
-      }
-    }
-
     const styles = useStyle({
       componentId: 'Icon' as const,
```

Comment:
Nitpic but this const doesn't do anything here, remove it please so it wont clutter the code